src/Products/Annotation/Controllers/AnnotationApiController.cs

```diff
@@ -509,81 +417,71 @@ public HttpResponseMessage Annotate(AnnotationPostedDataEntity annotateDocumentR
+                string tempPath = GetTempPath(documentGuid);
+
                 AnnotationDataEntity[] annotationsData = annotateDocumentRequest.annotationsData;
-                // initiate AnnotatedDocument object
                 // initiate list of annotations to add
-                List<AnnotationInfo> annotations = new List<AnnotationInfo>();
-              
-                DocumentInfoContainer documentInfo = AnnotationImageHandler.GetDocumentInfo(documentGuid, password);
-                // check if document type is image
-                if (SupportedImageFormats.Contains(Path.GetExtension(documentGuid)))
-                {
-                    documentType = "image";
-                }
-                // initiate annotator object  
-                string notSupportedMessage = "";
-                for (int i = 0; i < annotationsData.Length; i++)
+                List<AnnotationBase> annotations = new List<AnnotationBase>();
+
+                using (GroupDocs.Annotation.Annotator annotator = new GroupDocs.Annotation.Annotator(documentGuid, GetLoadOptions(password)))
                 {
-                    // create annotator
-                    AnnotationDataEntity annotationData = annotationsData[i];
-                    PageData pageData = documentInfo.Pages[annotationData.pageNumber - 1];
-                    // add annotation, if current annotation type isn't supported by the current document type it will be ignored
-                    try
+                    IDocumentInfo info = annotator.Document.GetDocumentInfo();
+
+                    for (int i = 0; i < annotationsData.Length; i++)
                     {
-                        BaseAnnotator annotator = AnnotatorFactory.createAnnotator(annotationData, pageData);
-                        if (annotator.IsSupported(documentType))
+                        AnnotationDataEntity annotationData = annotationsData[i];
+                        PageInfo pageInfo = info.PagesInfo[annotationsData[i].pageNumber - 1];
+                        // add annotation, if current annotation type isn't supported by the current document type it will be ignored
+                        try
                         {
-                            annotations.Add(annotator.GetAnnotationInfo(documentType));
+                            BaseAnnotator baseAnnotator = AnnotatorFactory.createAnnotator(annotationData, pageInfo);
+                            if (baseAnnotator.IsSupported(documentType))
+                            {
+                                annotations.Add(baseAnnotator.GetAnnotationBase(documentType));
+                            }
                         }
-                        else
+                        catch (Exception ex)
                         {
-                            notSupportedMessage = annotator.Message;
+                            throw new Exception(ex.Message, ex);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['System.Exception' should not be thrown by user code.](https://app.codacy.com/manual/GroupDocs/GroupDocs.Annotation-for-.NET-MVC/pullRequest?prid=5372470)